src/Servers/Kestrel/Core/src/Internal/Http/MessageBody.cs

```diff
@@ -231,49 +154,5 @@ protected void AddAndCheckConsumedBytes(long consumedBytes)
                 BadHttpRequestException.Throw(RequestRejectionReason.RequestBodyTooLarge);
             }
         }
-
-        private ValueTask<ReadResult> StartTimingReadAsync(CancellationToken cancellationToken)
-        {
-            var readAwaitable = _context.RequestBodyPipe.Reader.ReadAsync(cancellationToken);
```

Comment:
Start/StopTimingReadAsync is now basically duplicated 3 times. I would try to consolidate this. I know the line above to get the `ValueTask<ReadResult>` will be different in each implementation, but that can be fixed by calling an abstract method.